hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/BrokersClusterService.java

```diff
@@ -58,4 +72,34 @@ public boolean topicExists(Topic topic) {
     public boolean areOffsetsMoved(Topic topic, String subscriptionName) {
         return retransmissionService.areOffsetsMoved(topic, subscriptionName, clusterName);
     }
+
+    public boolean allSubscriptionsHaveConsumersAssigned(Topic topic, List<Subscription> subscriptions) {
+        List<String> consumerGroupsForSubscriptions = subscriptions.stream()
+                .map(sub -> kafkaNamesMapper.toConsumerGroupId(sub.getQualifiedName()).asString())
+                .collect(Collectors.toList());
+
+        try {
+            int requiredTotalNumberOfAssignments = numberOfPartitionsForTopic(topic) * subscriptions.size();
+            return numberOfAssignmentsForConsumersGroups(consumerGroupsForSubscriptions) == requiredTotalNumberOfAssignments;
+        } catch (Exception e) {
+            logger.error("Failed to check assignments for topic " + topic.getQualifiedName() + " subscriptions", e);
+            return false;
+        }
+    }
+
+    private int numberOfAssignmentsForConsumersGroups(List<String> consumerGroupsIds) throws ExecutionException, InterruptedException {
+        Collection<ConsumerGroupDescription> consumerGroupsDescriptions = adminClient.describeConsumerGroups(consumerGroupsIds).all().get().values();
+        Collection<MemberDescription> memberDescriptions = consumerGroupsDescriptions.stream().flatMap(desc -> desc.members().stream()).collect(Collectors.toList());
```

Comment:
we actually dont need to collect that, can be left as stream, minor

Reply:
done